### PR TITLE
Feature/rgd parser

### PR DESCRIPTION
**Pull request by nerdstrike**
## Description

RGDParser ported and adapted from core repo. I've added a few extra tests and improved the POD a bit for good measure

## Use case

Rat Xrefs

## Benefits

Happy rats

## Possible Drawbacks

Still adds unattached xrefs, a legacy behaviour.

## Testing

Tests created, found working. Should not have any impact on other code

### CODE REVIEW

File: modules/Bio/EnsEMBL/Xref/Parser/RGDParser.pm [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+    print "$count xrefs succesfully loaded and dependent on refseq\n";
+    print "$mismatch xrefs added but with NO dependencies\n";
+    print "$ensembl_count direct xrefs successfully loaded\n";
+    print "Tried to add $syn_count synonyms, including duplicates\n";
+  }
+  return;
+}
+
+# Predefined importance levels for the most valued RefSeq accession types
+my %refseq_priorities = (
```
Comment (premanand17) [inline modules/Bio/EnsEMBL/Xref/Parser/RGDParser.pm]: Why is this hash outside any method scope?

Comment (nerdstrike) [reply modules/Bio/EnsEMBL/Xref/Parser/RGDParser.pm]: Would you prefer it as a subroutine? Defined on construction? I put ReadOnly in front of it to make it slower? Put it at the top with a CAPS name so it's nowhere near where it gets invoked? Take your pick

Comment (premanand17) [reply modules/Bio/EnsEMBL/Xref/Parser/RGDParser.pm]: As it is used only in sort_refseq_accessions, I think it should go in to that routine?!

Comment (nerdstrike) [reply modules/Bio/EnsEMBL/Xref/Parser/RGDParser.pm]: and have it redefined every time it's used? It's run once per line in the file, i.e. lots

Comment (premanand17) [reply modules/Bio/EnsEMBL/Xref/Parser/RGDParser.pm]: Ok. Got it. Fair enough.

File: modules/t/rgd.t [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+  pass   => $config->{pass},
+  port   => $config->{port}
+);
+
+# Initialise two sources:
+my $rgd_source = $db->create_db_row('Source',{
+  name => 'RGD',
+  status => 'KNOWN',
+  priority_description => 'Test RGD source',
+  priority => 10
```
Comment (avullo) [inline modules/t/rgd.t]: Glad to see no ordered column.

File: modules/Bio/EnsEMBL/Xref/Parser/RGDParser.pm [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+sub run {
+
+  my ($self) = @_;
+  my $source_id    = $self->{source_id};
+  my $species_id   = $self->{species_id};
+  my $files        = $self->{files};
+  my $verbose      = $self->{verbose};
+  my $xref_dba     = $self->{xref_dba};
+
+  if((!defined $source_id) or (!defined $species_id) or (!defined $files) ){
```
Comment (avullo) [inline modules/Bio/EnsEMBL/Xref/Parser/RGDParser.pm]: I'd rather use 'unless' ANDing 'defined', it looks more readable to me, but it's a matter of preference.

File: modules/Bio/EnsEMBL/Xref/Parser/RGDParser.pm (hunk 1 above)
Comment (nerdstrike) [reply modules/Bio/EnsEMBL/Xref/Parser/RGDParser.pm]: I can make this change if you want:

`sub new {
  my ($class, @args) = @_;

  my $self = $class->SUPER::new(@args);
  # Set predefined importance levels for the most valued RefSeq accession types
  # 1 = most important
  $self->{refseq_priorities} = {
    NM => 1,
    NP => 1,
    NR => 1,
    XM => 2,
    XP => 2,
    XR => 2
  };
  return $self;
}`

Comment (avullo) [reply modules/Bio/EnsEMBL/Xref/Parser/RGDParser.pm]: Better but let's focus on other priorities.

File: modules/Bio/EnsEMBL/Xref/Parser/RGDParser.pm (hunk 3 above)
Comment (nerdstrike) [reply modules/Bio/EnsEMBL/Xref/Parser/RGDParser.pm]: I agree. Change made

File: modules/t/rgd.t [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+  Questions may also be sent to the Ensembl help desk at
+  <http://www.ensembl.org/Help/Contact>.
+
+=cut
+
+use strict;
+use warnings;
+use Test::More;
+use FindBin '$Bin';
+use Bio::EnsEMBL::Xref::Parser::RGDParser;
```
Comment (avullo) [inline modules/t/rgd.t]: Forgot to mention a preferred use_ok since we're talking about tests.

Comment (nerdstrike) [reply modules/t/rgd.t]: As you like

File: modules/t/rgd.t [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+});
+
+
+my $parser = Bio::EnsEMBL::Xref::Parser::RGDParser->new(
+  xref_dba => $xref_dba,
+  source_id => $rgd_source->source_id,
+  species_id => 34,
+  files => [ "$Bin/test-data/rgd.txt" ]
+);
+
```
Comment (avullo) [inline modules/t/rgd.t]: Since you're in the mood, how about isa_ok for completeness? ;)

**Comment by nerdstrike** [comment]
I'm done fiddling with this. Merge at discretion

**Pull request merged**